### ISSUE TITLE
Download large submodule issue

### ISSUE THREAD

**Issue by alibasheer** [wpbeirut/wp-beirut-boilerplate#1]
When I `git submodule update` it gets the 3 submodules we defined in the gitmodules file.
These submodules are full repos so we end up with downloading around 500MB
WordPress module: 178MB
WooCommerce module: 140MB
JetPack module: 203MB 

We need to find a way to get the latest version and not full repo

**Comment by FadiZahhar** [comment]
I found the cause,
infact when you update a submodule it will fetch all branches and tags and will fetch all the history of a repository.
I read a solution on stack overflow, but I didn't test it.
I found for the time beang to do a work around for this issue is as following.
instead of adding the submodule in the boilerplate for the update command, I recomand to follow the steps to add it for the first time, this will make sure that it will add the final release as cloning it, as example I tested on wordpress core and the size was 33 MG, also I tested on the plugin of woocommerce the size was 19MB .

**Comment by FadiZahhar** [comment]
this is the link on stack overflow
https://stackoverflow.com/questions/10914022/how-to-check-out-specific-version-of-a-submodule-using-git-submodule

I will update my workaround to fix this issue for now.

**Comment by FadiZahhar** [comment]
after doing several tests it is creating a submodule but take a lot of time , I'm replacing the concept of submoduling to regular cloning by demand, I updated the documentation, I will test it offline and see if it will take same time or not, in case yes, this is how wordpress github is optimized for the development. we will see and reply back.

**Comment by FadiZahhar** [comment]
after a back and forth, 
the best aproach was the follwoing.
I just removed the woocommerce and the jetpack on this release, and concentrate on the core of wordpress.
I downloded a clean optimized version of wordpress and add it as a wp-beirut-core repository , removing the wp-content and clean it to our use.
adding the cloned cleaned core repository as a submodule to our wp-beirut-boilerplate, it work like a charm. same command `git submodule init`, `git submodule update` will create the core in seconds.

this issue is closed.
